app/models/alchemy/element.rb

```diff
@@ -321,9 +321,11 @@ def copy_nested_elements_to(target_element)
     #   Pass an element name to get previous of this kind.
     #
     def previous_or_next(dir, name = nil)
-      elements = page.elements.published.where("#{self.class.table_name}.position #{dir} #{position}")
+      dir = dir == '>' ? '>' : '<'
+      order = dir == '>' ? :asc : :desc
```

Comment:
I don't think you need L324 then. When something else then `>` is given, the order is always `desc`. Should be safe enough.